### PR TITLE
tests: Add support for regex lookup in error msg

### PR DESCRIPTION
**Pull request by arno-lunarg**
Proposal for https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/8140

For now only tried it in NegativeGpuAVIndirectBuffer.DrawCount

### CODE REVIEW

File: tests/framework/error_monitor.h [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 #include "test_common.h"
 #include <unordered_set>
+#include <regex>
```
Comment (spencer-lunarg) [inline tests/framework/error_monitor.h:22]: this needs to be in the header? not in `error_monitor.cpp`?

**Review by spencer-lunarg — APPROVED** [review]
It looks good, one nit about the `include`

nice part is this is forward looking and won't break anything we already have

File: tests/unit/gpu_av_indirect_buffer.cpp [hunk 2]
```diff
@@ -241,7 +241,10 @@ TEST_F(NegativeGpuAVIndirectBuffer, DrawCount) {
     pipe.gp_ci_.layout = pipeline_layout.handle();
     pipe.CreateGraphicsPipeline();
 
-    m_errorMonitor->SetDesiredError("VUID-vkCmdDrawIndirectCount-countBuffer-03122");
+    m_errorMonitor->SetDesiredErrorRegex(
+        "VUID-vkCmdDrawIndirectCount-countBuffer-03122",
+        "Indirect draw count of 2 would exceed buffer size 16 of buffer VkBuffer 0x[a-f0-9]*\\[\\] stride = 16 offset = 0 "
```
Comment (spencer-lunarg) [inline tests/unit/gpu_av_indirect_buffer.cpp:246]: will be fun copying `0x[a-f0-9]*\\[\\] ` all over :laughing:

File: tests/framework/error_monitor.h (hunk 1 above)
Comment (arno-lunarg) [reply tests/framework/error_monitor.h:22]: yes, for `std::regex` definition

File: tests/unit/gpu_av_indirect_buffer.cpp (hunk 2 above)
Comment (arno-lunarg) [reply tests/unit/gpu_av_indirect_buffer.cpp:246]: oh I left that... honestly ` 0x[a-f0-9]*\\[\\] ` could just be `.*`

Comment (spencer-lunarg) [reply tests/unit/gpu_av_indirect_buffer.cpp:246]: ... that would feel nicer in my soul to copy in the future

File: tests/framework/error_monitor.h (hunk 1 above)
Comment (spencer-lunarg) [reply tests/framework/error_monitor.h:22]: missed that

File: tests/framework/error_monitor.cpp [hunk 3]
```diff
@@ -149,6 +165,12 @@ void ErrorMonitor::SetDesiredError(const char *msg, uint32_t count) {
     }
 }
 
+void ErrorMonitor::SetDesiredErrorRegex(const char *vuid, std::string msg_regex, uint32_t count /*= 1*/) {
+    for (uint32_t i = 0; i < count; i++) {
+        SetDesiredFailureMsgRegex(kErrorBit, vuid, msg_regex);
```
Comment (filnet) [inline tests/framework/error_monitor.cpp:170]: if count is greater than one then you might end up "compiling" the regex string multiple times.

Comment (spencer-lunarg) [reply tests/framework/error_monitor.cpp:170]: yes, I agree, will take a look later, for the short term, most things use count of `1` and the few GPU-AV tests that use it is `3` so we shouldn't accidentally hit a crazy regex compiling bottleneck, but I always monitor how long tests take to run regardless, so good thing to note as a possible spot to look at

Comment (spencer-lunarg) [reply tests/framework/error_monitor.cpp:170]: nvm, @arno-lunarg is a better man than me https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/8206

**Pull request merged**